### PR TITLE
Implement health checks for nflow-jetty.

### PR DESCRIPTION
**Pull request by jsyrjala**
Healtchecks are implemented using https://dropwizard.github.io/metrics/3.1.0/manual/healthchecks/
Also added servlet that provides a view to healt checks and metrics.

### CODE REVIEW

File: nflow-jetty/src/main/java/com/nitorcreations/nflow/jetty/config/NflowJettyConfiguration.java [hunk 1]
```diff
@@ -12,6 +16,10 @@
 import javax.ws.rs.core.Application;
 import javax.ws.rs.ext.RuntimeDelegate;
 
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.health.HealthCheckRegistry;
+import com.codahale.metrics.health.jvm.ThreadDeadlockHealthCheck;
+import com.codahale.metrics.jvm.*;
```
Comment (efonsell) [inline nflow-jetty/src/main/java/com/nitorcreations/nflow/jetty/config/NflowJettyConfiguration.java]: expand imports?

File: nflow-jetty/src/main/java/com/nitorcreations/nflow/jetty/config/NflowJettyConfiguration.java [hunk 2]
```diff
@@ -119,4 +130,29 @@ public JaxRsApiApplication jaxRsApiApplication() {
   public PlatformTransactionManager transactionManager(@NFlow DataSource nflowDataSource)  {
     return new DataSourceTransactionManager(nflowDataSource);
   }
+
+  @Bean
```
Comment (efonsell) [inline nflow-jetty/src/main/java/com/nitorcreations/nflow/jetty/config/NflowJettyConfiguration.java]: Move metrics beans to MetricsConfiguration

File: nflow-tests/src/test/java/com/nitorcreations/nflow/tests/MetricsAdminServletTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  @Test
+  public void t02_canFetchHealthChecks() throws IOException {
+    URI uri = URI.create("http://localhost:" + server.getPort() + "/metrics/healthcheck");
+    makeRequest(uri);
+  }
+
+  private void makeRequest(URI uri) throws IOException {
+    try(CloseableHttpClient httpclient = HttpClients.createDefault()) {
```
Comment (efonsell) [inline nflow-tests/src/test/java/com/nitorcreations/nflow/tests/MetricsAdminServletTest.java]: why not use jerseyclient here?

File: nflow-jetty/src/main/java/com/nitorcreations/nflow/jetty/config/NflowJettyConfiguration.java (hunk 2 above)
Comment (jsyrjala) [inline nflow-jetty/src/main/java/com/nitorcreations/nflow/jetty/config/NflowJettyConfiguration.java]: Done

File: nflow-tests/src/test/java/com/nitorcreations/nflow/tests/MetricsAdminServletTest.java (hunk 3 above)
Comment (jsyrjala) [inline nflow-tests/src/test/java/com/nitorcreations/nflow/tests/MetricsAdminServletTest.java]: Uses now jersey client, removed httpclient from pom.xml

**Pull request merged**